### PR TITLE
Use `dag_maker` fixture in test_backfill_job.py

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This change uses the dag_maker fixture in tests

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: tests/jobs/test_backfill_job.py [hunk 1]
```diff
@@ -71,14 +71,14 @@ def set_instance_attrs(self, dag_bag):
...
+    ):
 
-        with dag:
-            DummyOperator(task_id='op', pool=pool, task_concurrency=task_concurrency, dag=dag)
+        with dag_maker(dag_id=dag_id, schedule_interval='@daily', **kwargs) as dag:
+            DummyOperator(task_id=task_id, pool=pool, task_concurrency=task_concurrency)
 
-        dag.clear()
-        return dag
+        return dag, dag_maker.dag_run
```
Comment (uranusjr) [inline tests/jobs/test_backfill_job.py]: It might be possible to further simply the code by also turning this into a fixture.

```python
@pytest.fixture(scope="the_same_as_dag_maker")
def get_dummy_dag_and_run(dag_maker):
    def _get_dummy_dag_and_run(
        dag_id,
        pool=Pool.DEFAULT_POOL_NAME,
        task_concurrency=None,
        task_id='op',
        **kwargs,
    ):
        with dag_maker(dag_id=dag_id, schedule_interval='@daily', **kwargs) as dag:
            DummyOperator(task_id=task_id, pool=pool, task_concurrency=task_concurrency)
        return dag, dag_maker.dag_run

    return _get_dummy_dag_and_run
```

And use it like

```python
def test_unfinished_dag_runs_set_to_failed(self, get_dummy_dag_and_run):
    dag, dag_run = get_dummy_dag_and_run(dag_id='dummy_dag')

File: tests/jobs/test_backfill_job.py [hunk 2]
```diff
@@ -290,13 +306,13 @@ def test_backfill_conf(self):
 
         dr = DagRun.find(dag_id='test_backfill_conf')
 
-        assert conf_ == dr[0].conf
+        assert conf_ == dr[1].conf
```
Comment (ephraimbuddy) [inline tests/jobs/test_backfill_job.py]: Previously, the get_dummy_dag doesn't create dagrun. Now it creates dag_run, so when the  BackFillJob runs, it creates extra two dagruns which now have the conf. Escaping the first dagrun created by get_dummy_dag_and_run fixture, we now find the conf of the 2nd dagrun=>dr[1]

Comment (uranusjr) [reply tests/jobs/test_backfill_job.py]: Would it be easier to pass additional arguments to `find` to get only that second run? (e.g. `run_id`)

File: tests/jobs/test_backfill_job.py [hunk 3]
```diff
@@ -919,9 +895,9 @@ def test_backfill_max_limit_check_within_limit(self):
         assert 2 == len(dagruns)
         assert all(run.state == State.SUCCESS for run in dagruns)
 
-    def test_backfill_max_limit_check(self):
+    def test_backfill_max_limit_check(self, get_dag_test_max_active_limits):
         dag_id = 'test_backfill_max_limit_check'
-        run_id = 'test_dagrun'
+        run_id = 'test'
```
Comment (uranusjr) [inline tests/jobs/test_backfill_job.py]: Is this change needed? (just curious)

**Review by uranusjr — APPROVED** [review]

File: tests/jobs/test_backfill_job.py (hunk 3 above)
Comment (ephraimbuddy) [reply tests/jobs/test_backfill_job.py]: Checking why I did this...
